cli/pkg/local/server.go

```diff
@@ -90,6 +103,394 @@ func (s *Server) GetVersion(ctx context.Context, r *connect.Request[localv1.GetV
+	if err != nil {
+		if errors.Is(err, git.ErrRepositoryNotExists) {
+			isGithubRepo = false
+		} else if errors.Is(err, gitutil.ErrGitRemoteNotFound) {
+			githubRemoteFound = false
+		} else {
+			return nil, err
+		}
+	}
+
+	var hasUncommittedChanges *bool
+	if isGithubRepo && githubRemoteFound {
+		// check if there are uncommitted changes
+		// ignore errors since check is best effort and can fail in multiple cases
+		syncStatus, _ := gitutil.GetSyncStatus(s.app.ProjectPath, "", remote.Name)
+		hasUncommittedChanges = new(bool)
+		if syncStatus == gitutil.SyncStatusModified || syncStatus == gitutil.SyncStatusAhead {
+			*hasUncommittedChanges = true
+		} else {
+			*hasUncommittedChanges = false
+		}
+
+		// check git repo status, if git app installed and user is a collaborator and has authorised app on their account
+		repoStatus, err := c.GetGithubRepoStatus(ctx, &adminv1.GetGithubRepoStatusRequest{
+			GithubUrl: ghURL,
+		})
+		if err != nil {
+			return nil, err
+		}
+		repoAccess = repoStatus.HasAccess
+		if !repoStatus.HasAccess {
+			// we should have access to the repo before deploying
+			return connect.NewResponse(&localv1.DeployValidationResponse{
+				IsAuthenticated:         true,
+				LoginUrl:                loginURL,
+				IsGithubConnected:       true,
+				GithubGrantAccessUrl:    userStatus.GrantAccessUrl,
+				GithubUserName:          userStatus.Account,
+				GithubAppUserPermission: userStatus.UserInstallationPermission,
+				GithubOrganizationInstallationPermissions: userStatus.OrganizationInstallationPermissions,
+				IsGithubRepo:                  true,
+				IsGithubRemoteFound:           true,
+				IsGithubRepoAccessGranted:     repoAccess,
+				GithubUrl:                     "",
+				HasUncommittedChanges:         hasUncommittedChanges,
+				RillOrgExistsAsGithubUserName: false,
+				RillUserOrgs:                  nil,
+				LocalProjectName:              localProjectName,
+			}), nil
+		}
+	}
+
+	// get rill user orgs
+	resp, err := c.ListOrganizations(ctx, &adminv1.ListOrganizationsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	userOrgs := make([]string, 0, len(resp.Organizations))
+	for _, org := range resp.Organizations {
+		userOrgs = append(userOrgs, org.Name)
+	}
+	// TODO if len(userOrgs) > 0 then check if any project in these orgs already deploys from ghUrl, however I think this is not enough, probably should be checked globally ?
+
+	rillOrgExistsAsGitUserName := false
+	// if user does not any have orgs, check if any orgs exist as git username as we will suggest this for org name
+	if len(userOrgs) == 0 {
+		// check if any orgs exist as git username
+		_, err = c.GetOrganization(ctx, &adminv1.GetOrganizationRequest{
+			Name: userStatus.Account,
+		})
+		if err != nil {
+			if !errors.Is(err, database.ErrNotFound) {
+				return nil, err
+			}
+		} else {
+			rillOrgExistsAsGitUserName = true
+		}
+	}
+
+	// TODO check if any project exists with name localProjectName in any of the users rill org
```

Comment:
I don't think this is important, but what do you think?

Reply:
once we add retires with suffixes in case of conflict this is not that important